### PR TITLE
feat: cc-0010 add Swagger

### PR DESCRIPTION
**Pull request by GitHubEason**
feature CC-0010  Create swagger. The Swagger framework allows developers to create interactive, machine and human-readable API documentation, so it is easy for a front-end developer to know how to use this API.

### CODE REVIEW

File: app.ts [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
+import swaggerUi from "swagger-ui-express";
+
 import { logger } from "./src/utils/logger";
+import { swaggerDoc } from "./src/utils/swagger";
 
 const app = express();
 
 app.use(express.json());
 app.use(helmet());
 app.use(morgan("tiny"));
```
[the remark this answers is not in the document]
Comment (GitHubEason) [reply app.ts:15]: "tiny" is a predefined format in morgan, which is a parameter of morgan.

File: src/middleware/cors.ts [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 export const cors = (req: Request, res: Response, next: NextFunction) => {
-	res.setHeader("Access-Control-Allow-Origin", "*");
-	res.setHeader("Access-Control-Allow-Headers", "content-type");
-	res.setHeader("Access-Control-Allow-Methods", "*");
-	next();
+  res.setHeader("Access-Control-Allow-Origin", "*");
+  res.setHeader("Access-Control-Allow-Headers", "content-type");
+  res.setHeader("Access-Control-Allow-Methods", "*");
+  next();
 };
```
[the remark this answers is not in the document]
Comment (GitHubEason) [reply src/middleware/cors.ts:8]: Hi, Lawrence. Can we write cors by ourself in this stage pls?

File: app.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (LawrenceLiao) [reply app.ts:17]: All right, I misunderstood what this function does. It's fine to put "tiny" here, just a reminder that we can see not only one predefined string in morgan, you can take a loom and analyse which one is the most suitable for our project.

File: .env [hunk 3]
```diff
@@ -1,2 +1,3 @@
 PORT = 8080;
-API_PREFIX = /api/v1
\ No newline at end of file
+API_PREFIX = /api/v1
+yarn.lock
```
Comment (LawrenceLiao) [inline .env:3]: Why did you put yarn.lock in this file, this one should be only for environment variables, and the format isn't correct as well.

**Review by LawrenceLiao — APPROVED** [review]

File: .env [hunk 4]
```diff
@@ -1,3 +1,2 @@
 PORT = 8080;
```
Comment (Fuchih) [inline .env:1]: I suggest that:
 - no semi-colons
 - no spaces between equal sign

**Review by Fuchih — APPROVED** [review]
looks good to me

**Pull request merged**